packages/cli-vector/src/cli/cli.join.ts

```diff
@@ -13,21 +15,46 @@ async function fromFile(path: URL): Promise<URL[]> {
   const toProcess = await fsa.readJson(path);
   const paths: URL[] = [];
   if (!Array.isArray(toProcess)) throw new Error(`File ${path.href} is not an array`);
-  for (const tasks of toProcess) {
-    if (Array.isArray(tasks)) {
-      for (const task of tasks) {
-        if (typeof task !== 'string') throw new Error(`File ${path.href} is not an array of strings`);
-        paths.push(new URL(task));
-      }
-    } else if (typeof tasks === 'string') {
-      paths.push(new URL(tasks));
+  for (const task of toProcess) {
+    if (typeof task !== 'string') throw new Error(`File ${path.href} is not an array of strings`);
+    const sourceFile = new URL(task);
+    if (sourceFile.protocol === 'file:') {
+      paths.push(sourceFile);
     } else {
-      throw new Error(`File ${path.href} is not an array of strings`);
+      const fileName = basename(sourceFile.pathname);
+      if (fileName == null) throw new Error(`Unsupported source pathname ${sourceFile.pathname}`);
+      const localFile = new URL(fileName, fsa.toUrl('tmp/join/'));
+      const stream = fsa.readStream(sourceFile);
+      await fsa.write(localFile, stream);
+      paths.push(sourceFile);
```

Comment:
Shall we need push the `localFile` here? So we pass downloaded path into `tileJoin`.

Reply:
Cool, I think we can park this once the create command is merged, so we can test it later.